src/Core/AdminConsole/OrganizationFeatures/Policies/Implementations/PolicyRequirementQuery.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyRequirementQueries;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyRequirements;
+using Bit.Core.AdminConsole.Repositories;
+using Bit.Core.Models.Data.Organizations.OrganizationUsers;
+using Bit.Core.Settings;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.Policies.Implementations;
+
+public class PolicyRequirementQuery : IPolicyRequirementQuery
+{
+    private readonly IPolicyRepository _policyRepository;
+    private readonly PolicyRequirementRegistry _policyRequirements = new();
+
+    public PolicyRequirementQuery(IGlobalSettings globalSettings, IPolicyRepository policyRepository)
+    {
+        _policyRepository = policyRepository;
+
+        // Register Policy Requirement factory functions below
+        _policyRequirements.Add(SendPolicyRequirement.Create);
+        _policyRequirements.Add(up
+            => SsoPolicyRequirement.Create(up, globalSettings.Sso));
```

Comment:
Example of how you can just supply the factory function for simple requirements (which will be most of them), or inject additional values if needed (settings, feature flags, etc).

Reply:
Just an idea but maybe we could use Reflection here to find all classes that inherit `IPolicyRequirement` and automatically register them. Maybe it adds some unnecessary complexity though.

Reply:
I agree this would be too complex, and probably difficult for others to find/understand. My latest changes moved this to use the standard DI system which I think streamlines this a bit.